### PR TITLE
Add: DSA implementations for Win CNG and OpenSSL

### PR DESCRIPTION
**Pull request by mertakman**
This PR implements OpenSSL and Win CNG DSA implementations in all backends in Microsoft Go.

### CODE REVIEW

File: patches/0002-Add-crypto-backend-foundation.patch [hunk 1]
```diff
@@ -3,28 +3,31 @@ From: qmuntal <qmuntaldiaz@microsoft.com>
 Date: Thu, 30 Jun 2022 10:03:03 +0200
 Subject: [PATCH] Add crypto backend foundation
 
+fix:nobackend panic consistently without return
```
Comment (gdams) [inline patches/0002-Add-crypto-backend-foundation.patch]: remove incorrectly added line

```suggestion
```

File: patches/0002-Add-crypto-backend-foundation.patch [hunk 2]
```diff
@@ -3,28 +3,33 @@ From: qmuntal <qmuntaldiaz@microsoft.com>
...
```
Comment (gdams) [inline patches/0002-Add-crypto-backend-foundation.patch]: remove

File: patches/0002-Add-crypto-backend-foundation.patch [hunk 3]
```diff
@@ -3,28 +3,33 @@ From: qmuntal <qmuntaldiaz@microsoft.com>
...
+
+002fix
```
Comment (gdams) [inline patches/0002-Add-crypto-backend-foundation.patch]: remove

**Review by qmuntal — APPROVED** [review]
Looks good to me 🛩️

**Review by gdams — APPROVED** [review]

**Comment by qmuntal** [comment]
For https://github.com/microsoft/go/issues/1108.

**Pull request merged**